File: leetcode/167. 两数之和 II - 输入有序数组/solve_1.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &numbers, int target)
    {
        int left = 0, right = numbers.size() - 1;
        while (1)
        {
            int s = numbers[left] + numbers[right];
            if (s == target)
                break;
            if (s > target)
                --right;
            else
                ++left;
        }
        return {left + 1, right + 1};
    }
};

File: leetcode/LCR 006. 两数之和 II - 输入有序数组/solve_1.py
# 相向双指针
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers)-1
        while True: # left < right
            sum = numbers[left] + numbers[right]
            if (sum == target):
                break
            if (sum > target):
                # 大的减小
                right -= 1
            else:
                # 小的增大
                left += 1
        return [left,right]